src/jit/lsra.cpp

```diff
@@ -6709,6 +6785,13 @@ void LinearScan::processBlockStartLocations(BasicBlock* currentBlock, bool alloc
                     }
                     inVarToRegMap[assignedInterval->getVarIndex(compiler)] = REG_STK;
                 }
+#ifdef _TARGET_ARM_
+                // Consider overlapping floating point register for TYP_DOUBLE
+                else if (!assignedInterval->isConstant && isOverlappedRegRecord(assignedInterval, physRegRecord))
```

Comment:
You are right - I meant "its `assignedReg` IS `physRegRecord`". I think it would be good to rename and narrow the focus of the method.

---
In reply to: [112925890](https://github.com/dotnet/coreclr/pull/10972#discussion_r112925890) [](ancestors = 112925890)